This is my markdownfile for my portion of the final project. I will be working on Tables 2 and 3 from the paper Genome-Wide Comparative Analysis of Flowering-Related Genes in Arabidopsis, Wheat, and Barley by Peng et. al, 2015.

In [1]:
cd ..

C:\Users\Sarah\BCB546-Spring2022_Ice_Cream_Bean_SMinkler


In [6]:
conda install biopython

Solving environment: ...working... done

# All requested packages already installed.


Note: you may need to restart the kernel to use updated packages.


In [7]:
import pandas as pd

In [8]:
Supp_1_df = pd.read_excel (r'Supplemental_file_1[1].xls')

Opening Supplemental File 1 through Pandas in python

In [9]:
Supp_1_df

The 204 Arabidopsis flowering genes manually collected through literature and searches on TAIR (The Arabidopsis Information Resource)   \
0                                                Group                                                                                       
1                                           Autonomous                                                                                       
2                                           Autonomous                                                                                       
3                                           Autonomous                                                                                       
4                                           Autonomous                                                                                       
..                                                 ...                                                                                       
203                                           MicroRNA                                                                                       
204                                           MicroRNA                                                                                       
205                                                NaN                                                                                       
206                          * gene name not available                                                                                       
207  ** description related to flowering not available                                                                                       

    Unnamed: 1                                         Unnamed: 2  \
0        Locus                              Gene names (synonyms)   
1    AT2G19520  ACG1; ATMSI4; FVE; MSI4; MULTICOPY SUPPRESSOR ...   
2    AT2G43410                                                FPA   
3    AT3G04610                     FLK; FLOWERING LOCUS KH DOMAIN   
4    AT4G02560                                LD; LUMINIDEPENDENS   
..         ...                                                ...   
203  AT5G55835                              MICRORNA156H, MIR156H   
204  AT5G59505                              MICRORNA172E, MIR172E   
205        NaN                                                NaN   
206        NaN                                                NaN   
207        NaN                                                NaN   

                                            Unnamed: 3  
0                     Description related to flowering  
1                                    Control flowering  
2    Regulates flowering time in Arabidopsis via th...  
3           Positive regulation of flower development.  
4    Recessive mutants are late flowering. Expressi...  
..                                                 ...  
203  Encodes a microRNA that targets several SPL fa...  
204  Encodes a microRNA that targets several genes ...  
205                                                NaN  
206                                                NaN  
207                                                NaN  

[208 rows x 4 columns]

Checking to make sure I can open excel files properly, so tested with supplemental file one. Success.

In [10]:
Supp_3_df = pd.read_excel (r'Supplemental_file_3[1].xls')

In [11]:
Supp_3_df

The annotation of wheat flowering genes based on their top BLASTP hits in Arabidopsis  \
0                                               GeneID                                      
1                                  Traes_7DS_46E811D74                                      
2                                  Traes_4BS_2EE4988CD                                      
3                                  Traes_3AL_57252611A                                      
4                                  Traes_3DL_B63AF687B                                      
..                                                 ...                                      
896                                Traes_7DS_5527ABEE9                                      
897                                Traes_2AL_997B66BDC                                      
898                                Traes_3AL_8260356E3                                      
899                                Traes_7AS_5293F3028                                      
900                                Traes_6BL_B8497D305                                      

                     Unnamed: 1          Unnamed: 2  \
0            Ensembl annotation  Arabidopsis GeneID   
1    B-box transcription factor           AT3G02380   
2           DELLA protein RHT-1           AT1G14920   
3                DNA polymerase           AT5G67100   
4                DNA polymerase           AT5G67100   
..                          ...                 ...   
896     Uncharacterized protein           AT1G18100   
897     Uncharacterized protein           AT3G09220   
898     Uncharacterized protein           AT2G27990   
899     Uncharacterized protein           AT2G06005   
900     Uncharacterized protein           AT4G21200   

                                            Unnamed: 3 Unnamed: 4  
0                               Arabidopsis annotation    E-value  
1                        COL2, ATCOL2: CONSTANS-like 2        0.0  
2     GAI, RGA2: GRAS family transcription factor f...          0  
3                   ICU2: DNA-directed DNA polymerases          0  
4                   ICU2: DNA-directed DNA polymerases          0  
..                                                 ...        ...  
896   E12A11, MFT: PEBP (phosphatidylethanolamine-b...        0.0  
897                                    LAC7: laccase 7        0.0  
898                 BLH8, PNF: BEL1-like homeodomain 8        0.0  
899                FIP1: FRIGIDA interacting protein 1        0.0  
900          ATGA2OX8, GA2OX8: gibberellin 2-oxidase 8        0.0  

[901 rows x 5 columns]

Checking to make sure the file looks okay. A extra row has been added into the file, row 0.

In [12]:
pd.unique(Supp_3_df['Unnamed: 1'])

array(['Ensembl annotation', 'B-box transcription factor',
       'DELLA protein RHT-1', 'DNA polymerase',
       'Ent-copalyl diphosphate synthase', 'Flowering locus T',
       'Flowering Locus T-like protein, putative', 'GA2ox3-2 protein',
       'Gibberellin 20 oxidase 1-A', 'Gibberellin 20 oxidase 1-D',
       'Gibberellin 3-beta-dioxygenase 2-3',
       'Glucose-6-phosphate isomerase', 'Laccase',
       'MADS-box transcription factor',
       'MADS-box transcription factor TaAGL29',
       'MADS-box transcription factor TaAGL31',
       'MIKC-type MADS-box transcription factor WM27A',
       'Paf1 domain containing protein, expressed', 'Phytochrome',
       'R2R3-MYB protein', 'transcript:Traes_1BL_20ACD3421.1 ',
       'transcript:Traes_1BL_2DCDCD191.2 ',
       'transcript:Traes_2BS_510BCDA5A.1 ',
       'transcript:Traes_5DL_53D0B5EED.1 ',
       'transcript:Traes_7AS_722C376CF.2 ',
       'transcript:Traes_7AS_7DB94A25D.1 ',
       'transcript:Traes_7DS_F5FDB6AD1.2 ', 'Unchara

Checking the different values in the Ensembl annotation column. For this table, I am interested in the "uncharacterized proteins".

In [13]:
Supp_3_df['Unnamed: 1'].describe()

count                         901
unique                         28
top       Uncharacterized protein
freq                          859
Name: Unnamed: 1, dtype: object

Gathering basic info on column. The count should actually be 900, due to the extra column being counted.

In [14]:
Sorted_Ensembl_data = Supp_3_df.groupby('Unnamed: 1')

In [15]:
Sorted_Ensembl_data.describe()

The annotation of wheat flowering genes based on their top BLASTP hits in Arabidopsis  \
                                                                                                                              count   
Unnamed: 1                                                                                                                            
B-box transcription factor                                                                     1                                      
DELLA protein RHT-1                                                                            1                                      
DNA polymerase                                                                                 5                                      
Ensembl annotation                                                                             1                                      
Ent-copalyl diphosphate synthase                                                               1                                      
Flowering Locus T-like protein, putative                                                       1                                      
Flowering locus T                                                                              1                                      
GA2ox3-2 protein                                                                               1                                      
Gibberellin 20 oxidase 1-A                                                                     1                                      
Gibberellin 20 oxidase 1-D                                                                     1                                      
Gibberellin 3-beta-dioxygenase 2-3                                                             1                                      
Glucose-6-phosphate isomerase                                                                  6                                      
Laccase                                                                                        4                                      
MADS-box transcription factor                                                                  1                                      
MADS-box transcription factor TaAGL29                                                          1                                      
MADS-box transcription factor TaAGL31                                                          1                                      
MIKC-type MADS-box transcription factor WM27A                                                  1                                      
Paf1 domain containing protein, expressed                                                      1                                      
Phytochrome                                                                                    4                                      
R2R3-MYB protein                                                                               1                                      
Uncharacterized protein                                                                      859                                      
transcript:Traes_1BL_20ACD3421.1                                                               1                                      
transcript:Traes_1BL_2DCDCD191.2                                                               1                                      
transcript:Traes_2BS_510BCDA5A.1                                                               1                                      
transcript:Traes_5DL_53D0B5EED.1                                                               1                                      
transcript:Traes_7AS_722C376CF.2                                                               1                                      
transcript:Traes_7AS_7DB94A25D.1                                                               1                                   

Sorted data by Ensembl group so I can calculate the average of uncharacterized proteins. My results match that of the authors. 866 uncharacterized and 34 characterized proteins. 7 addtional proteins were added to the uncharacterized list because they had a transcript name, but no actual protein associated.

In [16]:
866/900

0.9622222222222222

In [17]:
34/900

0.03777777777777778

Calculating averages for wheat characterized and uncharacterized proteins. Characterized: 3.78%, Uncharacterized: 96.2%

In [19]:
Barley_Supp_3 = pd.read_excel (r'Supplemental_file_3[1].xls', sheet_name='Barley')
print (Barley_Supp_3)

    The annotation of barley flowering genes based on their top BLASTP hits in Arabidopsis  \
0                                               GeneID                                       
1                                           MLOC_66099                                       
2                                           MLOC_75886                                       
3                                           MLOC_44160                                       
4                                           MLOC_14118                                       
..                                                 ...                                       
271                                          MLOC_5926                                       
272                                          MLOC_5927                                       
273                                         MLOC_63361                                       
274                                          MLOC_4229      

Imported second sheet of excel file to look at Barley data. There are 275 total rows in this file. An additional row was added in the count.

In [20]:
Barley_Supp_3['Unnamed: 1'].describe()

count                         276
unique                         17
top       Uncharacterized protein
freq                          222
Name: Unnamed: 1, dtype: object

In [21]:
Sorted_Barley_data = Barley_Supp_3.groupby('Unnamed: 1')

In [22]:
Sorted_Barley_data.describe()

The annotation of barley flowering genes based on their top BLASTP hits in Arabidopsis  \
                                                                                                                               count   
Unnamed: 1                                                                                                                             
Bell-like homeodomain protein                                                                  1                                       
CBF protein 1                                                                                  1                                       
Centroradialis                                                                                 1                                       
Circadian clock associated 1                                                                   1                                       
DELLA protein SLN1                                                                             1                                       
DNA polymerase                                                                                 2                                       
ELF4-like protein                                                                              1                                       
Ensembl annotation                                                                             1                                       
FT                                                                                             1                                       
FT-like protein 4                                                                              1                                       
GID1-like gibberellin receptor                                                                 1                                       
Glucose-6-phosphate isomerase                                                                  2                                       
Homologous protein to TFL1                                                                     1                                       
MIKC-type MADS-box transcription factor WM27B                                                  1                                       
MIKC-type MADS-box transcription factor WM29B                                                  1                                       
Predicted protein                                                                             37                                       
Uncharacterized protein                                                                      222                                       

                                                                       \
                                              unique         top freq   
Unnamed: 1                                                              
Bell-like homeodomain protein                      1  MLOC_66099    1   
CBF protein 1                                      1  MLOC_75886    1   
Centroradialis                                     1  MLOC_44160    1   
Circadian clock associated 1                       1  MLOC_14118    1   
DELLA protein SLN1                                 1  MLOC_67940    1   
DNA polymerase                                     2  MLOC_50296    1   
ELF4-like protein                                  1  MLOC_58590    1   
Ensembl annotation                                 1      GeneID    1   
FT                                                 1  MLOC_68576    1   
FT-like protein 4                                  1  MLOC_74854    1   
GID1-like gibberellin receptor                     1   MLOC_6074    1   
Glucose-6-phosphate isomerase                      2   MLOC_1497    1   
Homologous protein to TFL1                         1  MLOC_35818    1   
MIKC-type MADS-box transcription factor WM27B      1  MLOC_65843    1   
MIKC-type MADS-box transcription factor WM29B      1   MLOC_5375    1   
Predicted p

###### There are a total of 222 Uncharacterized proteins and 37 predicted proteins, making the total number of unknown proteins 259. There were 16 identified proteins for Barley. This is different from what the authors obtained. They had 256 uncharacterized proteins and 16 characterized.

In [23]:
259/275

0.9418181818181818

In [27]:
16/275

0.05818181818181818

The percentages for each group are slightly off from what the authors obtained due to the differences in counts for each group.

Importing necessary functions to make Table 2.

In [36]:
pip install prettytable 

Note: you may need to restart the kernel to use updated packages.


In [37]:
from prettytable import PrettyTable

In [51]:
Table_2 = PrettyTable(["Ensembl annotation status", "Barley", "Wheat"])

In [52]:
Table_2.add_row(["Uncharacterized", "94.2 % (259)", "96.2 % (866)"])

In [53]:
Table_2.add_row(["Characterized", "5.8 % (16)", "3.8 % (34)"])

In [54]:
Table_2.add_row(["Total", "275", "900"])

Added all rows to the table

In [55]:
print(Table_2)

+---------------------------+--------------+--------------+
| Ensembl annotation status |    Barley    |    Wheat     |
+---------------------------+--------------+--------------+
|      Uncharacterized      | 94.2 % (259) | 96.2 % (866) |
|       Characterized       |  5.8 % (16)  |  3.8 % (34)  |
|           Total           |     275      |     900      |
+---------------------------+--------------+--------------+


In [57]:
data = Table_2.get_string()

In [59]:
with open('Table_2.txt', 'w') as f: f.write(data)

Table 2 has been made into a file and is located on the repository for viewing